GooglePlayPlugins/com.google.play.billing/Samples/TrivialKart/scripts/Controller/Game/GamePlayCanvasController.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using UnityEngine;
+using UnityEngine.UI;
+
+/// <summary>
+/// Controller for the game play canvas.
+/// It updates the coin text indicator in the play canvas.
+/// </summary>
+public class GamePlayCanvasController : MonoBehaviour
+{
+    public Text coinsCount;
+
+    private bool _canvasHasStarted = false;
```

Comment:
curious why do you need this?

Reply:
When start the game, the game data will load at Awake() function in gameManager.cs, which happens at the same time with the OnEnble() function. Therefore, there maybe a case that we refresh the game play canvas when the game data is not loaded. I added this bool to avoid this problem.